apps/fhir/server/authentication.py

```diff
@@ -8,30 +8,33 @@
 logger = logging.getLogger('hhs_server.%s' % __name__)
 
 
-def authenticate_crosswalk(crosswalk):
+def match_hicn_hash(hicn_hash):
```

Comment:
Test cases:
- happy case
- fhir_id not found
- 500 from server
- too many entries
- `total` not present in response (but multiple entries)
- malformed response